scripts/qa_crowbarsetup.sh

```diff
@@ -3343,6 +3343,89 @@ function addfloatingip
     nova floating-ip-create | tee floating-ip-create.out
     floatingip=$(perl -ne "if(/\d+\.\d+\.\d+\.\d+/){print \$&}" floating-ip-create.out)
     nova add-floating-ip "$instanceid" "$floatingip"
+    echo $floatingip
+}
+
+function oncontroller_nova_evacuate
+{
+    # Test nova evacuate
+    local vm_name=NovaEvacuate
+
+    # JSON parser with python
+    local PJSON='import json,sys; l1=json.load(sys.stdin)'
+
+    # Get hypervisor hostname
+    echo $(openstack hypervisor list -c 'Hypervisor Hostname'  --format json) > json
+    local hypervisor_host_fqdn=$(cat json | python -c "$PJSON; print l1[0]['Hypervisor Hostname']")
+    local hypervisor_host=$(echo $hypervisor_host_fqdn | cut -d '.' -f 1)
+    # Crate instance on specific hypervisor image always exist after tempest
+    # run (cirros-0.3.4-x86_64-tempest-machine)
+    echo $hypervisor_host
+    echo $vm_name
+    nova boot --image cirros-0.3.4-x86_64-tempest-machine --flavor tempest-stuff --availability-zone nova:$hypervisor_host $vm_name
+    echo "Wait for instance launch"
+    sleep 60
+    # Create floating ip assign to instance
+    addfloatingip $vm_name
+    local floatingip=$(openstack server show -c addresses  --format value $vm_name | cut -d " " -f 2)
+    echo "Floating ip created" $floatingip
+    # Update security group for icmp
+    nova secgroup-add-rule default icmp -1 -1 0.0.0.0/0 &> /dev/null
+
+    # Check if instance is ACTIVE and Running
+    ping -c 10 $floatingip >/dev/null 2>&1
+    local ping_result=$?
+    echo "Instance ping res" $ping_result
+    local status=$(openstack server show -c status --format value $vm_name)
+    echo $status
+    if [ "$status" == "ACTIVE" ] && [ $ping_resul=0 ] ; then
+        echo "Instance launched successfully"
+    else
+        echo "Instance launch failed"
+        exit 1
+    fi
+
+    # Simulate failure of hypervisor node
+    kill_pacemaker_remote $hypervisor_host_fqdn $floatingip
+    echo "Pacemaker killed"
+    local ret=0
+
+    # Check that instance evacuated to working hypervisor host
+    local hypervisor_evacuate_host=$(openstack server show -c OS-EXT-SRV-ATTR:hypervisor_hostname --format value $vm_name)
+    if [ "$hypervisor_evacuate_host" != "$hypervisor_host" ] ; then
+        echo "Instance evacuated"
+    else
+        echo  "Evacuate fail, instance remains on failed host"
+        ret=20
+    fi
+
+    exit $ret
+}
+
+function kill_pacemaker_remote
+{
+    local hypervisor_fqdn=$1
+    local floatingip=$2
+
+    # Simulate failure of hypervisor node
+    echo $hypervisor_fqdn
+    ssh -t -T  $hypervisor_fqdn 'killall -9 pacemaker_remoted'
+    local MSG_LOG="/var/log/messages"
+    local ret=1
+    local n=300
+    while [ $n -gt 0 ] && [ "$ret" != "0" ]; do
+    #cat $MSG_LOG | grep -qe $SEARCH_PATTERN
+    tail -f -n0 $MSG_LOG | grep 'Completed evacuation of'
```

Comment:
tail -f will not return, unless you used grep -q behind the pipe which might exit on first match and then close, which triggers a SIGPIPE in tail.